sympy/algebras/quaternion.py

```diff
@@ -328,6 +328,15 @@ def pow(self, p):
             p = p >> 1
             q = q * q
 
+        if p < -1:
+            r = abs(p)
+            while r > 0:
+                if r & 1:
+                    res = q.inverse() * res
+
+                r = r >> 1
+                q.inverse() = q.inverse() * q.inverse()
```

Comment:
I don't think this is a valid operation. Assigning to a method is not valid (atleast not in python).